docs/reference/commandline/run.md

```diff
@@ -242,27 +242,50 @@ For the `overlay2` storage driver, the size option is only available if the
 backing fs is `xfs` and mounted with the `pquota` mount option.
 Under these conditions, user can pass any size less than the backing fs size.
 
-### Mount tmpfs (--tmpfs)
+### Mount tmpfs (--mount type=tmpfs)
+
+> **Note**: The `--tmpfs` flag is deprecated.
 
 ```bash
-$ docker run -d --tmpfs /run:rw,noexec,nosuid,size=65536k my_image
+$ docker run --rm -it \
+  --mount type=tmpfs,tmpfs-size=65536k,tmpfs-mode=600,dst=/run \
+  ubuntu:latest
 ```
 
-The `--tmpfs` flag mounts an empty tmpfs into the container with the `rw`,
-`noexec`, `nosuid`, `size=65536k` options.
+This example mounts an empty file 65536k in size into `/run` inside the
+container, with mode 600 (`rw-------`). Because the type of `tmpfs`, the file's
+contents of the file are never written to a block device within the container
```

Comment:
IIRC, `tmpfs` is still being written to disk if there's insufficient memory, and memory is swapped to disk. Only `ramfs` is never written to disk (which is used for secrets and configs, but not supported by `--mount`)